pkgs/by-name/ze/zenmap/package.nix

```
@@ -0,0 +1,78 @@
+{
+  gobject-introspection,
+  gtk3,
+  lib,
+  nmap,
+  python3Packages,
+  wrapGAppsHook3,
+  xterm,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "zenmap";
+  version = nmap.version;
+  pyproject = true;
+
+  src = nmap.src;
+
+  prePatch = ''
+    cd zenmap
+  '';
+
+  build-system = with python3Packages; [
+    setuptools
+  ];
+
+  buildInputs = [
+    nmap
+    gtk3
+    xterm
+  ];
+
+  nativeBuildInputs = [
+    wrapGAppsHook3
+    gobject-introspection
+  ];
+
+  nativeCheckInputs = [
+    nmap
+  ];
+
+  dependencies = with python3Packages; [
+    pygobject3
+  ];
+
+  postInstall = ''
+    wrapProgram "$out/bin/zenmap" \
+      --prefix PATH : ${lib.makeBinPath [ nmap ]}
```

```suggestion
  dontWrapGApps = true;

  postInstall = ''
    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
    makeWrapperArgs+=(--prefix PATH : ${lib.makeBinPath [ nmap ]})
```
Otherwise you get this mess of wrappers :wink:  (Python + GApps + your PATH wrapper)
```
arne@nixOS> \ls -lah /nix/store/7sr0cnai7pwxcrnkxhppaw1x64iaasf9-zenmap-7.95/bin                                                                                                                                              ~/nixpkgs-wt-2
total 100K
dr-xr-xr-x 1 root root  266 1970-01-01 01:00 .
dr-xr-xr-x 1 root root   44 1970-01-01 01:00 ..
-r-xr-xr-x 1 root root  20K 1970-01-01 01:00 su-to-zenmap.sh
-r-xr-xr-x 1 root root 1,9K 1970-01-01 01:00 .su-to-zenmap.sh-wrapped
-r-xr-xr-x 1 root root  20K 1970-01-01 01:00 zenmap
-r-xr-xr-x 1 root root  20K 1970-01-01 01:00 .zenmap-wrapped
-r-xr-xr-x 1 root root  16K 1970-01-01 01:00 .zenmap-wrapped_
-r-xr-xr-x 1 root root  16K 1970-01-01 01:00 ..zenmap-wrapped-wrapped
-r-xr-xr-x 1 root root  780 1970-01-01 01:00 ...zenmap-wrapped-wrapped-wrapped
```